bindings/java/devdoc/java_out_of_process_hld.md

```diff
@@ -0,0 +1,122 @@
+Java Out of Process Module API
```

Comment:
Your diagram doesn't cover the full module API. You are missing parseJsonConfig^ and freeConfig^.

*^ not actual names*